From 81a8ff3743ae3dd21b5a0225f1ecf6bc798cbad4 Mon Sep 17 00:00:00 2001
From: Tapani <tapani@vmail.me>
Date: Fri, 17 May 2013 14:50:09 +0800
Subject: [PATCH 214/225] Add support for AUO 97G070 LVDS panel

---
 drivers/video/mxc/ldb.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/video/mxc/ldb.c b/drivers/video/mxc/ldb.c
index ad39a89..8b8e6bc 100644
--- a/drivers/video/mxc/ldb.c
+++ b/drivers/video/mxc/ldb.c
@@ -107,6 +107,15 @@ static u8 g_edid[2][512];
 
 static struct fb_videomode ldb_modedb[] = {
 	{
+	/* 800x480 @ 60 Hz , pixel clk @ 32MHz */
+	 "LDB-WVGA", 60, 800, 480, 29850, 
+	 89, 164, 
+	 23, 10, 
+	 10, 10,
+	 FB_SYNC_CLK_LAT_FALL,
+	 FB_VMODE_NONINTERLACED,
+	 0,},
+	{
 	 "LDB-WXGA", 60, 1280, 800, 14065,
 	 40, 40,
 	 10, 3,
-- 
1.7.10.4

