From d31524ac67e07c2a6a0ec871f6081afd8aff1890 Mon Sep 17 00:00:00 2001
From: Tapani <tapani@vmail.me>
Date: Tue, 21 May 2013 11:27:02 +0800
Subject: [PATCH 218/223] Fix signal type for AUO 97G070 LVDS panel

---
 drivers/video/mxc/ldb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/mxc/ldb.c b/drivers/video/mxc/ldb.c
index 8b8e6bc..475c3a3 100644
--- a/drivers/video/mxc/ldb.c
+++ b/drivers/video/mxc/ldb.c
@@ -112,7 +112,7 @@ static struct fb_videomode ldb_modedb[] = {
 	 89, 164, 
 	 23, 10, 
 	 10, 10,
-	 FB_SYNC_CLK_LAT_FALL,
+	 FB_SYNC_HOR_HIGH_ACT,
 	 FB_VMODE_NONINTERLACED,
 	 0,},
 	{
-- 
1.7.10.4

