From 6dfece682f761c772d3c4f62a7036c6619d7e602 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 10 Apr 2013 09:38:30 -0500
Subject: [PATCH 8/8] deb: disable header_check/header_install

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 scripts/package/builddeb |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index a5f197b..d29ea76 100644
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -126,10 +126,10 @@ if grep -q '^CONFIG_MODULES=y' .config ; then
 	fi
 fi
 
-if [ "$ARCH" != "um" ]; then
-	$MAKE headers_check KBUILD_SRC=
-	$MAKE headers_install KBUILD_SRC= INSTALL_HDR_PATH="$libc_headers_dir/usr"
-fi
+#if [ "$ARCH" != "um" ]; then
+#	$MAKE headers_check KBUILD_SRC=
+#	$MAKE headers_install KBUILD_SRC= INSTALL_HDR_PATH="$libc_headers_dir/usr"
+#fi
 
 # Install the maintainer scripts
 # Note: hook scripts under /etc/kernel are also executed by official Debian
-- 
1.7.10.4

